### ISSUE TITLE
Bazel 4.2.2 crashes when adding arguments to run_shell

### ISSUE THREAD

**Issue by EmilePapillon** [bazelbuild/bazel#14378]
### Description of the problem / feature request:

Adding Args type arguments to run_shell causes Bazel to crash. 

According to the documentation Args should be valid values for arguments to the run_shell function. 

`Args are constructed by calling ctx.actions.args(). They can be passed as the arguments parameter of ctx.actions.run() or ctx.actions.run_shell()`

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Create the following rule in WORKSPACE/crash_reproduction/crash.bzl:
```
def _crash_impl(ctx):
    output=ctx.actions.declare_file("output.txt")
    args = ctx.actions.args()
    args.add("byebye_bazel")
    args.add(output.path)
    CMD = """
echo $1 > $2
"""
    ctx.actions.run_shell(
        arguments=args,
        command=CMD,
        outputs=[output]
    )

crash = rule(
    implementation = _crash_impl
)
```
And the following BUILD file to crash_reproduction package:
```
load("//crash_reproduction:crash.bzl", "crash")

crash(
	name="crash_me"
)
```
Then build with `bazel build //crash_reproduction:crash_me`

Note that I already have a workaround without using Args, but I wanted to report this finding. 

### What operating system are you running Bazel on?

Ubuntu 20.04 (5.4.0-91-generic #102-Ubuntu SMP)

### What's the output of `bazel info release`?

release 4.2.2

###  Have you found anything relevant by searching the web?

No

### Any other information, logs, or outputs that you want to share?

Here is the stack trace:

FATAL: bazel crashed due to an internal error. Printing stack trace:
java.lang.RuntimeException: Unrecoverable error while evaluating node 'ConfiguredTargetKey{label=//crash_reproduction:foo, config=BuildConfigurationValue.Key[96d6638d77e294d5ea45abfa4bdf12d435d0d7fb38e15d839ac9c6ef83541e29]}' (requested by nodes )
        at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:563)
        at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:398)
        at java.base/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(Unknown Source)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
        at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: net.starlark.java.eval.Starlark$UncheckedEvalException: java.lang.ClassCastException: class com.google.devtools.build.lib.analysis.starlark.Args$MutableArgs cannot be cast to class net.starlark.java.eval.Sequence (com.google.devtools.build.lib.analysis.starlark.Args$MutableArgs and net.starlark.java.eval.Sequence are in unnamed module of loader 'app') (Starlark stack: [_crash_impl@/home/emile/tmp/bazel_tut/crash_reproduction/crash.bzl:9:26, run_shell@<builtin>])
        at net.starlark.java.eval.Starlark.fastcall(Starlark.java:610)
        at net.starlark.java.eval.Eval.evalCall(Eval.java:641)
        at net.starlark.java.eval.Eval.eval(Eval.java:460)
        at net.starlark.java.eval.Eval.exec(Eval.java:264)
        at net.starlark.java.eval.Eval.execStatements(Eval.java:81)
        at net.starlark.java.eval.Eval.execFunctionBody(Eval.java:65)
        at net.starlark.java.eval.StarlarkFunction.fastcall(StarlarkFunction.java:160)
        at net.starlark.java.eval.Starlark.fastcall(Starlark.java:606)
        at com.google.devtools.build.lib.analysis.starlark.StarlarkRuleConfiguredTargetUtil.buildRule(StarlarkRuleConfiguredTargetUtil.java:130)
        at com.google.devtools.build.lib.analysis.ConfiguredTargetFactory.createRule(ConfiguredTargetFactory.java:369)
        at com.google.devtools.build.lib.analysis.ConfiguredTargetFactory.createConfiguredTarget(ConfiguredTargetFactory.java:195)
        at com.google.devtools.build.lib.skyframe.SkyframeBuildView.createConfiguredTarget(SkyframeBuildView.java:940)
        at com.google.devtools.build.lib.skyframe.ConfiguredTargetFunction.createConfiguredTarget(ConfiguredTargetFunction.java:1031)
        at com.google.devtools.build.lib.skyframe.ConfiguredTargetFunction.compute(ConfiguredTargetFunction.java:371)
        at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:477)
        ... 7 more
Caused by: java.lang.ClassCastException: class com.google.devtools.build.lib.analysis.starlark.Args$MutableArgs cannot be cast to class net.starlark.java.eval.Sequence (com.google.devtools.build.lib.analysis.starlark.Args$MutableArgs and net.starlark.java.eval.Sequence are in unnamed module of loader 'app')
        at com.google.devtools.build.lib.analysis.starlark.StarlarkActionFactory.runShell(StarlarkActionFactory.java:424)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at net.starlark.java.eval.MethodDescriptor.call(MethodDescriptor.java:135)
        at net.starlark.java.eval.BuiltinFunction.fastcall(BuiltinFunction.java:77)
        at net.starlark.java.eval.Starlark.fastcall(Starlark.java:606)
        ... 21 more

**Comment by meteorcloudy** [comment]
Can you confirm if this issue also exists in 4.2.1?

**Comment by hxtk** [comment]
> Can you confirm if this issue also exists in 4.2.1?

I just found this issue by googling because I encountered the same problem in my own code, and I am using Bazel 4.2.1.

**Comment by tetromino** [comment]
This was fixed by 8a43e0cb973ab979da8140bc31fd94651537ce5b. Unfortunately, that commit is not in any stable release yet; you'd have to switch to 5.0.0-pre or 6.0.0-pre to get the fix.
